Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/Follow.cs

```diff
@@ -0,0 +1,573 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Solvers
+{
+    /// <summary>
+    /// Follow solver positions an element relative in front of the forward axis of the reference.
+    /// The element can be loosly constrained (a.k.a. tag-along) so that it doesn't follow until it is too far.
+    /// </summary>
+    public class Follow : Solver
+    {
+        [SerializeField]
+        [Tooltip("Position lerp multiplier")]
+        private float moveToDefaultDistanceLerpTime = 0.1f;
+
+        /// <summary>
+        /// Position lerp multiplier.
+        /// </summary>
+        public float MoveToDefaultDistanceLerpTime
+        {
+            get { return moveToDefaultDistanceLerpTime; }
+            set { moveToDefaultDistanceLerpTime = value; }
+        }
+        
+        [SerializeField]
+        [Tooltip("The desired orientation of this object")]
+        private SolverOrientationType orientationType = SolverOrientationType.MaintainGoal;
+
+        /// <summary>
+        /// The desired orientation of this object.
+        /// </summary>
+        public SolverOrientationType OrientationType
+        {
+            get { return orientationType; }
+            set { orientationType = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("Min distance from eye to position element around, i.e. the sphere radius")]
+        private float minDistance = 1f;
+
+        /// <summary>
+        /// Min distance from eye to position element around, i.e. the sphere radius.
```

Comment:
We need to avoid using eye or head. Solvers should be able to function with different tracked transforms